### ISSUE TITLE
Authorino set up in Istio as a Auth GRPC service

### ISSUE THREAD

**Issue by maleck13** [Kuadrant/kuadrant-controller#149]
**What**

When installing Authorino, the Kuadrant Operator should also configure Isito to know about the location of the Auth service. 
It will need to know the namespace and possible the name of the configmap to to edit. 
Permissions wise this is a bit tricky because it could be installed in any namespace so we will need a cluster role possibly we can limit which named resource it can edit if the configmap is always named the same way

**Comment by eguzki** [comment]
I have been working on this and here my notes.

When istio is [installed with istioctl](https://istio.io/latest/docs/setup/install/istioctl/), there is no operator to reconciliate configuration deployment. As it is specified in the [doc](https://istio.io/latest/docs/tasks/security/authorization/authz-custom/#define-the-external-authorizer), you need to manually update the `istio-system/istio` config map and then restart istiod to allow the change to take effect. Does not feel right that the kuadrant operator restart istiod as it does not manage it and may not even have permissions to do so. So, I tried with [installing Istio with the Istio operator](https://istio.io/latest/docs/setup/install/operator/).

I have deployed the istio operator and Istio mesh with the `IstioOperator` custom resource
```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  name: istiocontrolplane
  namespace: istio-system
spec:
  components:
    base:
      enabled: true
    cni:
      enabled: false
    egressGateways:
      - enabled: false
        name: istio-egressgateway
    ingressGateways:
      - enabled: true
        name: istio-ingressgateway
    pilot:
      enabled: true
  namespace: istio-system
  profile: default
  tag: 1.12.1
  values:
    gateways:
      istio-ingressgateway:
        autoscaleEnabled: false
        type: ClusterIP
    global:
      istioNamespace: istio-system
    pilot:
      autoscaleEnabled: false
```
Istio instance in the cluster up&running is given from kuadrant's perspective. So, the kuadrant's operator is expected to update the IstioOperator CR to define the external authorizer, Authorino. Manually I updated the CR to add the meshConfig

```yaml
meshConfig:
    extensionProviders:
      - envoyExtAuthzGrpc:
          port: 50051
          service: authorino-authorino-authorization.kuadrant-system.svc.cluster.local
        name: kuadrant-authorization
```

Unfortunately, the Istio operator did not reconcile the change and the `istio` config map did not get the changes nor the istiod was restarted. 

So there is no a clean way to do this from the kuadrant's operator. The kuadrant operator does not even know the name of the IstioOperator CR. For the test carried out, I knew the name because I was deploying Istio but in a production environment, the name of the IstioOperator CR is unkown. Or even worse, there might not be any Istio Operator up&running. 

Feels like setting up external authorizer in the existing istio instance is out of scope of the kuadrant's operator responsabilitites. 

fyi cc @maleck13 @rahulanand16nov 

any feedback would be welcome

**Comment by rahulanand16nov** [comment]
Not sure but I remember hearing that reconciling `configmap` usually takes a lot of time by Istio and usually people restart `istiod` to fasten the process. Are you sure, it's not reconciling even after some time?

**Comment by eguzki** [comment]
Thanks for the tip. Unfortunately, it was not a time issue. However, good news :partying_face: . 

I dug into the istio components and I noticed that the operator log was reporting some weird error. After some investigation, the issue was fixed in Istio 1.13. So, I upgraded istio to 1.13 and the operator was not blocked by any error and successfully reconciliated the external authorizer config into the required configmap. Furthermore, istiod was not restarted and yet the configuration was applied. I checked and verified that authorino was getting requests for authorization. 

PS: For the record, the issue with Istio 1.12.1 was [this](https://github.com/istio/istio/issues/36114) and this [answer](https://github.com/istio/istio/issues/36114#issuecomment-1096348714) made it. Answer was made 28 days ago :)

I am going to implement it in the kuadrant operator. The remaining issue is about the name of the `IsitioOperator` resource name. I will read it from an env var and give a default value `istiocontrolplane`. Besides, the implementation requires istio to be managed by their operator.
